lib/xform.js

```diff
@@ -41,18 +58,164 @@ var XFormState = State.extend(function (self, name, opts) {
 
     // Defaults
     opts = _.defaults(opts || {}, {
-        next:null,
+        next: null,
+        xforms_service_url: 'prd-vumi-xforms.aws.prk-host.net',
+        xforms_service_error_message: 'Error contacting the xforms service',
     });
 
     State.call(self, name, opts);
 
     // Store options
     self.next = opts.next;
+    self.xforms_service_url = opts.xforms_service_url;
+    self.xforms_service_error_message = opts.xforms_service_error_message;
+
+    if(opts.xform !== undefined) {
+        self.xform = opts.xform;
+    } else {
+        throw new Error("opts.xform must be defined for XFormState");
```

Comment:
We should probably extend `vumigo_v02.utils.BaseError` to make `XformsStateError` and raise that so that people don't have to match the error string to tell what sort of error happened.